ERCS/erc-0626.md

```diff
@@ -0,0 +1,119 @@
+
+## Specification
+
+The contract interface includes the following methods:
+
+### Read Methods
+
+1. **`json()`**
+
+   - **Inputs:** None.
+   - **Output:** `string` (a JSON object as a string)
+   - **Description:** Returns the JSON object in its latest state. The output uses the most recent value for each key in the JSON object, mimicking a REST JSON call.
+   - **Behavior:** Each key in the JSON object reflects the latest "Last In, First Out" (LIFO) state of its value.
+
+2. **`version(string key)`**
+
+   - **Inputs:** 
+     - `key` (`string`): The JSON key whose version history is requested.
+   - **Output:** `string` (a JSON array as a string)
+   - **Description:** Returns an array of all versions of the specified key's value in JSON format. 
+     The array is ordered chronologically, with the earliest version at index 0 and the most recent version at the highest index.
+
+### Write Method
+
+1. **`write(string[] keys, string[] values, bool replace)`**
+
+   - **Inputs:**
+     - `keys` (`string[]`): Array of JSON key strings, where each key represents a path within the JSON object.
+     - `values` (`string[]`): Array of JSON-compatible strings corresponding to each key. Can represent primitive types, arrays, or objects.
+     - `replace` (`bool`): Determines whether to replace an existing key's latest value.
+       - If `replace` is `true` and the key exists, the existing value is updated in the `json` output.
+       - However, the previous value persists in the `version` output, with the new value appended at the end of the version history array.
+       - If `replace` is `false` and the key already exists, the transaction reverts.
+   - **Output:** None.
+   - **Description:** Writes new values to the JSON object, either adding a new key or updating an existing one, based on the `replace` parameter.
+
+#### Additional Requirements
+
+- **Universal ABI:** To support widespread adoption and ease of use, the ABI should be standardized for REST-like abstraction within popular Web3 libraries. 
+  By making this contract accessible through Web3 abstractions, developers with REST experience can seamlessly interact with the contract.
+  
+## Rationale
+
+1. **REST-like Access via JSON Method**  
+   The `json` method enables developers to interact with the contract as if it were a RESTful API, improving accessibility for those familiar with traditional web development paradigms.
+
+2. **Version Management via Version Method**  
+   The `version` method provides a straightforward version control system for each key, offering a history of values that developers can reference without altering the main JSON structure. 
+   This method maintains immutability for historical values while allowing updates to be appended.
+
+3. **Compatibility with Web3 Abstractions**  
+   Ensuring a simple and standardized ABI is essential for usability with Web3 libraries, thus enhancing developer experience and facilitating onboarding.
+
+## Example Usage
```

Comment:
The bot will complain about this section. You should move it to be a subheading under Specification.